share/wake/lib/system/remote_cache_runner.wake

```diff
@@ -106,168 +267,11 @@ export def mkRemoteCacheRunner (rscApi: RemoteCacheApi) (hashFn: Result RunnerIn
-                            # would already exist on disk
-                            "-f",
-                            path,
-                            link,
-                        )
-                        Nil
-                        | setPlanLabel "rsc: symlink {link} to {path}"
-                        | setPlanPersistence Once
-                        | runJobWith localRunner
-                        | setJobInspectVisibilityHidden
-                        | isJobOk
-                    else failWithError "rsc: Failed to link {link} to {path}"
-
-                    Pass Unit
-
-                def outputFilesDownload =
-                    outputFiles
-                    | map doDownload
-
-                # Symlinks don't need to wait for files, the symlinks will just momentarily be invalid if created first
-                def outputSymlinksDownload =
-                    outputSymlinks
-                    | map doMakeSymlink
-
-                def resolvedOutputs =
-                    outputFiles
-                    | map getCacheSearchOutputFilePath
-
-                def resolvedSymlinks =
-                    outputSymlinks
-                    | map getCacheSearchOutputSymlinkLink
-
-                def resolvedDirectories =
-                    outputDirs
-                    | map getCacheSearchOutputDirectoryPath
-
-                def outputs = resolvedOutputs ++ resolvedDirectories ++ resolvedSymlinks
-                def predict = Usage status runtime cputime mem ibytes obytes
-
-                require Pass stdout =
-                    stdoutDownload
-                    | addErrorContext "rsc: Failed to download stdout for '{label}'"
-
-                require Pass stderr =
-                    stderrDownload
-                    | addErrorContext "rsc: Failed to download stderr for '{label}'"
-
-                def _ = virtual job stdout stderr status runtime cputime mem ibytes obytes
-                def inputs = map getPathName (input.getRunnerInputVisible)
-
-                # We don't actually care about the result here but we need to ensure that all
-                # downloads have completed before returning.
-                require Pass _ =
-                    outputFilesDownload
-                    | findFail
-                    | addErrorContext "rsc: Failed to download a blob"
-
-                require Pass _ =
-                    outputSymlinksDownload
-                    | findFail
-                    | addErrorContext "rsc: Failed to create a symlink"
-
-                Pass (RunnerOutput inputs outputs predict)
+                match (rehydrateJob response label input job)
+                    Pass x -> Pass x
+                    # If the job hydration fails for any reason just run the job as normal.
+                    # There is no point in attempting to push since the server just said its cached
+                    Fail _ -> baseDoIt job (Pass input)
```

Comment:
Should we not call `runJobAndUpload` here? Or do we think that will be problematic if the job is currently in a state where its matching cache response can't be rehydrated?

Reply:
It would be strictly wasted work. Since we are inside the "match" case then the server already thinks it has the job. If we then go and repost it we'll just get rejected by the `allowed` route

Not a whole lot we can do in this case but maybe would be nice for the client to somehow tell the server that the specific job is acting up